aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/request/AWSS3StorageUploadInputStreamRequest.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage.s3.request;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.amplifyframework.storage.StorageAccessLevel;
+import com.amplifyframework.storage.s3.ServerSideEncryption;
+
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Parameters to provide to S3 that describe a request to upload a inputStream.
+ */
+public final class AWSS3StorageUploadInputStreamRequest {
```

Comment:
There's a lot of duplication in this class with `AWSS3StorageUploadFileRequest`.  Could you create a common abstract super class, `AWSS3StorageUploadRequest`, which contains everything that is common to both?

Reply:
Might make sense to do this for the `AWSS3StorageUploadInputStreamOperation`, as well. There could be an `AWSS3StorageUploadOperation` that gets extended in two different ways.